src/modules/uORB/Subscription.cpp

```diff
@@ -88,4 +88,29 @@ void Subscription::unsubscribe()
 	_last_generation = 0;
 }
 
+bool Subscription::ChangeInstance(uint8_t instance)
+{
+	if (instance != _instance) {
+		DeviceMaster *device_master = uORB::Manager::get_instance()->get_device_master();
+
+		if (device_master != nullptr) {
+			if (!device_master->deviceNodeExists(_orb_id, _instance)) {
+				return false;
+			}
```

Comment:
Any reason why you're not allowing to change to a not-yet-existing instance?

Reply:
No, that's leftover garbage from an earlier attempt that no longer makes sense. Removing.

Reply:
Actually, now I recall it was more of a concern with it being a potentially valid instance and thinking forward to a time when the max is variable per topic.

At the moment I think I'll keep it to only allow you to change to a currently valid instance and actually check everywhere.

Reply:
Alright, let's keep an eye on how usage and other needs continue to evolve.